app/src/main/java/com/hedvig/app/feature/genericauth/CreateOtpAttemptUseCase.kt

```diff
@@ -0,0 +1,29 @@
+package com.hedvig.app.feature.genericauth
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.CreateOtpAttemptMutation
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+
+class CreateOtpAttemptUseCase(
+    private val apolloClient: ApolloClient,
+) {
+    sealed class Result {
+        data class Success(
+            val token: String,
+        ) : Result()
+
+        object Error : Result()
+    }
+
+    suspend operator fun invoke(email: String) = when (
+        val result = apolloClient
+            .mutate(CreateOtpAttemptMutation(email = email))
+            .safeQuery()
+    ) {
+        is QueryResult.Success -> {
+            Result.Success(result.data.login_createOtpAttempt)
+        }
+        else -> Result.Error
```

Comment:
Get error message here

Reply:
The error-message isn't user-presentable, right? That was my motivation for not extracting it. Perhaps logging it could be valid, though?

Reply:
In `<T> ApolloCall<T>.safeQuery()` we currently get the exception message or error message from the response. We should map these error states to user readable strings though, but for now I think its better to go with what we have. It makes debugging so much easier.

Reply:
Or we could just map this error to a general network error string :) And log the exception message. That probably makes more sense.